runtime/src/validate_block/implementation.rs

```diff
@@ -56,14 +57,14 @@ trait Storage {
 #[cfg(not(feature = "std"))]
 #[doc(hidden)]
 pub fn validate_block<B: BlockT, E: ExecuteBlock<B>>(
-	mut block_data: &[u8],
+	mut arguments: &[u8],
 ) {
-	use codec::Decode;
-
-	let block_data = crate::ParachainBlockData::<B>::decode(&mut block_data)
+	let (parent_hash, block_data): (B::Hash, crate::ParachainBlockData::<B>) = Decode::decode(&mut arguments)
 		.expect("Could not decode parachain block.");
 	// TODO: Add `PolkadotInherent`.
 	let block = B::new(block_data.header, block_data.extrinsics);
+	assert!(parent_hash == *block.header().parent_hash(), "Invalid parent hash");
```

Comment:
I'm not sure about error reporting here. Should `validate_block` signature be changed?

Reply:
`assert!` is okay here.